FreeRTOS/FreeRTOS-Kernel
# Fix MISRA_C_2012 rule 8.4 violation

Fix MISRA C 2012 rule 8.4 violation

Description
-----------
MISRA rule 8.4 

> A compatible declaration shall be visible when an object or function with external linkage is defined.

**MISRA Rule Violation**

This rule states that if no available declaration is present for a variable in a header file, it should be either declared as static variable or an appropriate declaration be assigned.

2 errors are suppressed at [1](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/queue.c#L168) and [2](https://github.com/FreeRTOS/FreeRTOS-Kernel/blob/main/tasks.c#L439) by comment, because:

1. `xQueueRegistry` is utilized as an extern variable in queue unit tests.
2. A declaration in header file is not useful as the assembly code will still need to declare it separately from the port files.

Test Steps
-----------
N/A.

Checklist:
----------




By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
